### PR TITLE
GFORMS-1199 File upload and delete for Non JS users restores data from file-upload envelope metadata

### PR DESCRIPTION
**Pull request by girishkamat**

### CODE REVIEW

**Review by josef-vlach — CHANGES_REQUESTED** [review]
There seems to be duplicate data created every upload. 
```json
"fields": [
    {
      "id": "fileA",
      "value": ""
    },
    {
      "id": "textA",
      "value": ""
    },
    {
      "id": "textA",
      "value": "F1"
    },
    {
      "id": "textA",
      "value": "F2"
    }
  ],
  ```
In combination of leaving the form and then coming back later, it seems to put stale data to the form.

It would be nice handle dups in form data.

**Comment by girishkamat** [comment]
The issue with duplicate has been fixed. Well spotted!

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: public/javascripts/gformFileUpload.js [hunk 1]
```diff
@@ -369,23 +369,13 @@
...
-          deleteUrl +
-          '" class="govuk-link" data-form-component-id="' +
-          formComponentId +
-          '" data-form-id="' +
-          formTemplateId +
-          '" data-access-code="' +
-          accessCode +
-          '"><span aria-hidden="true">' +
+          " <button type='submit' class='link' data-module='govuk-button' id='fileDelete' formaction='/submissions/form/delete-file/" +
+          formTemplateId + "/-/" + formComponentId + "'> " +
```
[the remark this answers is not in the document]
Comment (girishkamat) [reply public/javascripts/gformFileUpload.js]: Thanks, its been fixed now

File: app/uk/gov/hmrc/gform/views/form/snippets/uploaded_files.scala.html [hunk 2]
```diff
@@ -33,16 +33,10 @@
...
-        data-form-component-id="@formComponentId.value"
-        data-form-id="@templateId"
-        data-access-code="@accessCode">
-        <span aria-hidden="true">@messages("file.delete")</span>
-        <span class="govuk-visually-hidden">@messages("file.delete.file", currentValue)</span>
-      </a>
+      <span>@{currentValue.replace(fileId.value + "_", "")} @govukButton(Button(content = Text(messages("file.delete")),
+          inputType=Some("submit"),
+          attributes = Map("id" -> "fileDelete", "formaction" -> uk.gov.hmrc.gform.fileupload.routes.FileUploadController.deleteFile(formTemplateId, maybeAccessCode, formComponentId).path),
+          classes="govuk-button link"))</span>
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/views/form/snippets/uploaded_files.scala.html:39]: This seems to be affecting accessibility. There is no information what file is this delete button removing. Or do I miss something?

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/views/form/snippets/uploaded_files.scala.html]: Thanks. Fixed both in template and javascript using aria-label

**Review by SandipKhedekar — APPROVED** [review]

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**